language-support/ts/codegen/tests/BUILD.bazel

```diff
@@ -10,23 +10,65 @@ load("@os_info//:os_info.bzl", "is_windows")
 # directory. It is referenced from the 'build-and-lint' test that
 # follows.
 daml_compile(
-    name = "daml2ts-test",
+    name = "build-and-lint-pkg",
     srcs = glob(["daml/**/*.daml"]),
     main_src = "daml/Main.daml",
 )
 
-# The 'build-and-lint' (test) target:
-#  - Causes 'daml2ts-test.dar' to be produced (from the sources under 'ts/daml');
-#  - Invokes 'daml2ts' on that dar to produce typescript bindings in' ts/generated/src/daml';
-#  - Invokes 'yarn install' in 'ts' to install dependencies;
-#  - Invokes 'yarn build' then 'yarn lint' on each of the '@daml/types', '@daml/ledger' and 'generated' packages;
-#  - Invokes 'yarn test' from the 'ts/generated' directory.
-# That last step causes 'ts/generated/src/tests/__tests__/test.ts' to be executed which:
-#  - Spins up a sandbox running the 'daml2ts-test.dar';
+# build-and-lint
+# ==============
+#
+# 'build-and-lint.sh' is an integration test. It builds a DAR from
+#  some DAML, invokes daml2ts on the DAR, uploads the DAR to a ledger
+#  and then runs tests on the TypeScript generated by daml2ts via a
+#  json-api connected to the ledger.
+#
+#  The main inputs to 'build-and-lint' are:
+#     - Bazel artifacts : '@daml/types' and '@daml/ledger' (these are
+#       npm packages);
+#     - Source artifacts:
+#       - 'ts/codegen/tests/daml/*'
+#         The DAML that goes into the DAR ('build-and-lint-pkg.dar');
+#       - 'ts/codegen/tests/ts/*'
+#         A skeleton yarn workspaces root containing a TypeScript package
+#         'build-and-lint' which houses the test driver 'test.ts'.
+#
+# 'build-and-lint' creates a temp directory ('root' say), and uses the
+# above inputs (via filesystem copies, damlc, daml2ts and yarn install)
+# to produce a filesystem with the following structure:
+#
+#   root/
+#      build-and-lint/
+#        src/
+#          __tests__/
+#            test.ts
+#      daml2ts/
+#        build-and-lint-pkg/  -- The TypeScript generated from 'build-and-lint-pkg.dar'
+#        057eed/
+#        ...
+#      package.json  -- Enumeration of the workspaces : build-and-lint-pkg, daml2ts/057eed, ...
+#      node_modules/
+#        @daml/
+#          types/
+#          ledger/
+#        ...
+#
+# In more detail, the actions of 'build-and-lint' are to:
+#  - Produce 'build-and-lint-pkg.dar';
```

Comment:
I don’t think this is true. The DAR is built separately and you pass it in via `data`.

Reply:
You are right.